ports/stm32/mpbthciport.c

```diff
@@ -140,26 +137,18 @@ int mp_bluetooth_hci_uart_write(const uint8_t *buf, size_t len) {
 // Callback to copy data into local hci_uart_rx_buf_data buffer for subsequent use.
 STATIC int mp_bluetooth_hci_uart_msg_cb(void *env, const uint8_t *buf, size_t len) {
     (void)env;
-    if (hci_uart_rx_buf_len + len > MP_ARRAY_SIZE(hci_uart_rx_buf_data)) {
-        len = MP_ARRAY_SIZE(hci_uart_rx_buf_data) - hci_uart_rx_buf_len;
+    mp_bluetooth_hci_uart_readchar_t handler = env;
+    for (size_t i = 0; i < len; i++) {
+        handler(buf[i]);
     }
-    memcpy(hci_uart_rx_buf_data + hci_uart_rx_buf_len, buf, len);
-    hci_uart_rx_buf_len += len;
+    processed += len;
     return 0;
 }
 
-int mp_bluetooth_hci_uart_readchar(void) {
-    if (hci_uart_rx_buf_cur >= hci_uart_rx_buf_len) {
-        hci_uart_rx_buf_cur = 0;
-        hci_uart_rx_buf_len = 0;
-        rfcore_ble_check_msg(mp_bluetooth_hci_uart_msg_cb, NULL);
-    }
-
-    if (hci_uart_rx_buf_cur < hci_uart_rx_buf_len) {
-        return hci_uart_rx_buf_data[hci_uart_rx_buf_cur++];
-    } else {
-        return -1;
-    }
+int mp_bluetooth_hci_uart_readpacket(mp_bluetooth_hci_uart_readchar_t handler) {
+    processed = 0;
+    rfcore_ble_check_msg(mp_bluetooth_hci_uart_msg_cb, (void *)handler);
```

Comment:
Can you update `rfcore_ble_check_msg` to return size_t (the total number of bytes processed). Then we don't need the global.

Reply:
Ah yes, I kept forgetting that rfcore is ours (as opposed to vendor code) and could be udpated to do things like this.